src/coreclr/jit/flowgraph.cpp

```diff
@@ -8632,13 +8632,29 @@ void Compiler::fgAddReversePInvokeEnterExit()
     varDsc->lvType      = TYP_BLK;
     varDsc->lvExactSize = eeGetEEInfo()->sizeOfReversePInvokeFrame;
 
-    GenTree* tree;
-
     // Add enter pinvoke exit callout at the start of prolog
 
-    tree = gtNewOperNode(GT_ADDR, TYP_I_IMPL, gtNewLclvNode(lvaReversePInvokeFrameVar, TYP_BLK));
+    GenTree* pInvokeFrameVar = gtNewOperNode(GT_ADDR, TYP_I_IMPL, gtNewLclvNode(lvaReversePInvokeFrameVar, TYP_BLK));
+
+    GenTree* stubArgument;
+
+    if (info.compPublishStubParam)
+    {
+        // If we have a secret param for a Reverse P/Invoke, that means that we are in an IL stub.
+        // In this case, the method handle we pass down to the Reverse P/Invoke helper should be
+        // the target method, which is passed in the secret parameter.
+        stubArgument = gtNewLclvNode(lvaStubArgumentVar, TYP_I_IMPL);
+    }
+    else
+    {
+        stubArgument = gtNewIconNode(0, TYP_I_IMPL);
```

Comment:
why does this argument do?

Reply:
This argument allows us to resolve the target MethodDesc when we're compiling a Reverse P/Invoke IL stub (in comparison to an UnmanagedCallersOnly function)

Reply:
This particular case is when we are not in a stub, so we pass a null handle to the helper to signify that.